crates/matrix-sdk-crypto/src/file_encryption/attachments.rs

```diff
@@ -277,12 +277,13 @@ impl<'a, R: Read + ?Sized + 'a> AttachmentEncryptor<'a, R> {
 
 /// Struct holding all the information that is needed to decrypt an encrypted
 /// file.
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
The only sane way would be for `AttachmentDecryptor` to accept a `&MediaEncryptedInfo` instead of a `MediaEncryptedInfo`. I'm not sure it's necessary for `AttachmentDecryptor` to take ownership of the media encrypted info, except if we want to “destroy” the secret. Thoughts?

Reply:
Hmm seems like you are right. We can remove the zeroization from the decryptor and move it into the `MediaEncryptionInfo` `drop()` implementation.

Reply:
Done.